### DISCUSSION TITLE
CuteleeView templates default include path

### DISCUSSION THREAD

**Discussion by MikePooh** [cutelyst/cutelyst#325] [General]
In [Cutelyst/Plugins/View/Cutelee/cuteleeview.cpp](https://github.com/cutelyst/cutelyst/blob/master/Cutelyst/Plugins/View/Cutelee/cuteleeview.cpp) CuteleeView constructor, row#74, there is

`setIncludePaths({ QDir::currentPath() });`

My application on the target machine is starting by parent process.

According to [QtDocs](https://doc.qt.io/qt-5/qdir.html#currentPath)
_Returns the absolute path of the application's current directory. The current directory is the last directory set with QDir::setCurrent() or, if that was never called, the **directory at which this application was started at by the parent process.**_

So my `QDir::currentPath()` was different from `QCoreApplication::applicationDirPath()` namely folder with the execuatble.

I spent couple of hours to recognize why I get "Template not found error" regardless I have templates in application directory right near the executable file.

I purpose to use `QCoreApplication::applicationDirPath()` as default templates path instead of `QDir::currentPath()`, or both of them. If @dantti accept this I can make a PR. If not, I can just switch `QDir::setCurrent(path)` to the path of executable.

**Comment by dantti** [comment]
That's an else case, which should normally not used, CutelystView should most of the time have a Cutelyst::Application parent, and you can always overwrite that manually, but having both will cause no harm tho...

**Reply by MikePooh** [reply]
Hmm.. Very strange. My CutelystView has parent of WebApp which is inherited from Cutelyst::Application

webapp.h
```
class WebApp : public Cutelyst::Application
{
    Q_OBJECT
    ...
public:
   bool init() override;
   ...
}
```

webapp.cpp
```
WebApp::WebApp(QObject *parent) : Cutelyst::Application(parent)

bool WebApp::init()
{
   ...
  new CuteleeView(this);
}
```

So CuteleeView constructor received the parent which is QObject and inherited from Cutelyst::Application
I've insert a couple of qDebugs() just before the mentioned if/else construction in cuteleeview.cpp:

```
    auto app = qobject_cast<Application *>(parent);
    qDebug() << "parent" << parent;
    qDebug() << "static_cast" << static_cast<Application*>(parent);
    qDebug() << "qobject_cast" << qobject_cast<Application*>(parent);
    qDebug() << "dynamic_cast" << dynamic_cast<Application*>(parent);
```
And get the output
```
parent WebApp(0x5593ab4e8810)
static_cast WebApp(0x5593ab4e8810)
qobject_cast QObject(0x0)
dynamic_cast WebApp(0x5593ab4e8810)
```
qobject_cast is 0 that's why I've dropped to else. Why so if my WebApp class is successor of Cutelyst::Application?

**Reply by MikePooh** [reply]
I've found that qobject_cast failed beacuse of CUTELYST_APPLICATION marco was omitted in WebApp class.
